lib/datasource/go/index.ts

```diff
@@ -68,12 +69,24 @@ async function getDatasource(goModule: string): Promise<DataSource | null> {
           .replace(/\/$/, ''),
       };
     }
-    const gitlabRes = gitlabRegExp.exec(goSourceUrl);
-    if (gitlabRes) {
+    const gitlabUrl = gitlabHttpsRegExp.exec(goSourceUrl);
+    const gitlabModule = gitlabRegExp.exec(goModule);
+    const gitlabUrlName = gitlabUrl?.[2].replace(/\/$/, '');
+    const gitlabModuleName = gitlabModule?.[2].replace(/\/$/, '');
```

Comment:
This would be quite hard to maintain in future. I know it wasn't too easy before either, but now it's getting even more complex. Could you use regex group matching instead (e.g. to replace the `[2]` indexes) plus commenting if necessary?